# Project Description

In principle, microarray and RNA-Seq measure the same thing: the genome-wide abundance of mRNA molecules. However, the technologies are fundamentally different and the results from the same experiments assayed using the different methods can show substantial differences. Assessing the differences in gene expression measurement exclusively due to the underlying technology requires careful and comprehensive study design, such that other sources of variation in gene expression are controlled. The goal of the study was to characterize the concordance of differential gene expression across platforms, test and compare how effective each platform is at detecting expected pathway-level effects based on each treatment’s MOA, and assess the MOA prediction accuracy of each platform using a test set. <br /><br />

In this project, we will reproduce the results from the paper, [A comprehensive study design reveals treatment- and transcript abundance–dependent concordance between RNA-seq and microarray data](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4243706/), Figure 2a and Figure 3b+c, as well as compare the pathway enrichment results reported in the paper.

# Contributors

Data Curator: [Neha Gupta](https://github.com/neha163)<br />
Programmer: [Rachel Thomas](https://github.com/RaePayne03)<br />
Analyst: [Yichi Zhang](https://github.com/MogicianEik)<br />
Biologist: [Varun Raghuraman](https://github.com/vrvarun)<br />

# Repository Contents

1. RNA-Seq Sample Statistics and Alignment.<br />
2. Read counting with featureCounts.<br />
3. RNA-Seq Differential Expression with DESeq2.<br />
4. Microarray Differential Expressoin with Limma.<br />
5. Concordance between microarray and RNA-Seq DE genes. <br />
6. A report in details.
